script.js

```diff
@@ -3,29 +3,62 @@ var mainPalette = {
   buttonContainer: document.querySelector('.button-container'),
   savedPaletteContainer: document.querySelector('.saved-palettes-container'),
   currentPalette: '',
+  savedPalettes: [],
+
+  newPaletteButton() {
+      this.currentPalette.refresh();
+      this.displayMain();
+  },
+
+  savePaletteButton() {
+    if (this.savedPalettes.length === 8) {
+      this.savedPalettes.pop();
+    }
+    this.savedPalettes.unshift(this.currentPalette);
+    this.resetLocks();
+    this.createNew();
+    this.displaySaved();
+  },
 
   createNew() {
       this.currentPalette = new Palette();
-      this.display()
+      this.displayMain();
   },
 
-  newPaletteButton() {
-      this.currentPalette.refresh();
-      this.display()
+  resetLocks() {
+    this.currentPalette.palette.forEach((color) => {
+      color.locked = false;
+    });
+  },
+  
+  deletePalette(paletteId) {
+    this.savedPalettes.splice(paletteId, 1);
+    this.displaySaved();
   },
 
-  savePaletteButton() {
-    console.log("save works")
+  displaySaved() {
+    var smallPalettes = '';
+    this.savedPalettes.forEach((object, index) =>{
+     smallPalettes += `<section class="container saved-palette" data-palette-number="${index}">
+        <section class="swatch-small" style="background-color:${object.palette[0].hexCode};" ></section>
```

Comment:
I would stay away from using sections inside of sections like this. It can be hard to troubleshoot. HTML elements can be tricky to decide which to use where and why, but I advise staying away from this practice. This is mainly for developer empathy, but can also play a big role in accessibility.

A div would be perfectly acceptable to use instead of the children sections. In reality, they're both flavorless elements. Just make sure to apply the appropriate labels when doing so. But this is me being a little nit-picky :)